### ISSUE TITLE
[Backport v0.8] Fleet Reports Wrong Helm Version, Can't Deploy Traefik Chart [SURE-6813]

### ISSUE THREAD

**Issue by rancherbot** [rancher/fleet#1819]
This is a backport issue for https://github.com/rancher/fleet/issues/1643, automatically created via rancherbot by @manno

Original issue description:

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

Have been using fleet to deploy Traefik for almost a year. Have just rebuilt my cluster and connected my Git account. When I try to deploy Traefik shows the following error:

ErrApplied(1) [Cluster fleet-local/local: execution error at (traefik/templates/deployment.yaml:3:8): ERROR: Helm >= 3.9.0 is required]

### Expected Behavior

Should deploy without error.

`helm version
version.BuildInfo{Version:"v3.11.3-rancher1", GitCommit:"8853d6f9875eb3138d10f4af7a4a67130039e28b", GitTreeState:"clean", GoVersion:"go1.19.4"}`

### Steps To Reproduce

Try to deploy Traefik using Fleet.

### Environment

```markdown
- Architecture: x86
- Fleet Version: 2.7.5
- Cluster:
  - Provider: K3S
  - Options: 8 nodes, Longhorn, Metallb
  - Kubernetes Version: 1.25.9
```

### Logs

```markdown
ErrApplied(1) [Cluster fleet-local/local: execution error at (traefik/templates/deployment.yaml:3:8): ERROR: Helm >= 3.9.0 is required]
```

**Comment by manno** [comment]
copied from https://github.com/rancher/fleet/issues/1643#issuecomment-1764854185

## Additional QA
 
### Problem

Fleet didn't provide all ["Capabilities"](https://helm.sh/docs/chart_template_guide/builtin_objects/) to helm charts it deployed. For the helm version it provided an old default.

### Solution

Update how we use the helm SDK.
 
### Testing

A helm chart can now use
`{{ .Capabilities.HelmVersion.Version }}` in templates. One popular helm chart that does that is "traefik". 



### Engineering Testing
#### Manual Testing

Tested that the helm version values are now present.

#### Automated Testing

There is an integration test to make sure the capabilities are filled.

* `{{ .Capabilities.KubeVersion.Version }}`
* `{{ join \", \" .Capabilities.APIVersions |  }}`
* `{{ .Capabilities.HelmVersion.Version }}`

### QA Testing Considerations
 
-

#### Regressions Considerations

-

**Comment by mmartin24** [comment]
> ### System information:
> * Rancher version: 2.7.11-head (fc2c10a)
> * Fleet version: v0.8.2-rc.3
> 
> Test case documented in https://app.qase.io/case/FLEET-94
> ### Rancher Upgrade Scenario
> * [ ]  Deployed single node k3s (`v1.26.10+k3s2`) cluster  `2.7.10`
> * [ ]  Deployed traefik chart on local cluster. Issue could be reproduced
![repro_error_ok](https://github.com/rancher/fleet/assets/37271841/1be55797-2bb3-4832-b4d6-f1c0ad6a8f07)
> * [ ]   Upgrade from Rancher `2.7.10` to `2.7.11-head (fc2c10a)` and error is gone (Force Update may help)
![image](https://github.com/rancher/fleet/assets/37271841/8ae293bb-ada8-4219-b3e7-327ccff5cdb8)

> ### Upgraded Kubernetes version after deployment
 * [ ]  Upgraded deployment to  k3s version (`v1.27.6+k3s1`) on deployed cluster and upgrade was possible

> ### Fresh Rancher installation
> * [ ]   Fresh Rancher installation with k3s `v1.26.10+k3s2` directly on `2.7.11-head (fc2c10a)` and later traefik chart installed as described ok
> * [ ]   Fresh Rancher installation with k3s `v1.27.1+k3s1` directly on `2.7.11-head (fc2c10a)` and later traefik chart installed as described not possible due to error already mentioned [here](https://github.com/rancher/fleet/issues/2084#issuecomment-1941232477)

Error:
```
Error: INSTALLATION FAILED: chart requires kubeVersion: < 1.27.0-0 which is incompatible with Kubernetes v1.27.6+k3s1

**Comment by thehejik** [comment]
Verified as fixed in `rancher:v2.7.11-alpha3`:
![Image](https://github.com/rancher/fleet/assets/12828077/976a8dad-c62c-4602-8850-98df2fc89ad2)
